x/liquidity/spec/01_concepts.md

```diff
@@ -4,88 +4,60 @@
+## Equivalent Swap Price Model (ESPM)
 
-These features of the liquidity module create incentives to transfer tokens:
+The liquidity module is a Cosmos SDK implementation of an AMM system with a novel economic model called the Equivalent Swap Price Model (ESPM). The key distinguishing feature of the ESPM model from the Constant Product Market Maker (CPMM) model (for example, Uniswap) is the implementation of a hybrid system. This system combines an orderbook model exchange with a simple liquidity pool model that governs the order book with a set of order rules and performs execution in batches. In the ESPM, the pool price is always equal to the last swap price which reduces opportunities for arbitrage.
 
-- Combines a traditional order book-based exchange system with a Uniswap-like AMM mechanism. This hybrid system deepens liquidity for the token swap marketplace.
+The ESPM model is intended to provide protection against price volatility, transaction ordering vulnerabilities, and losses due to arbitrage. AMMs such as Uniswap do not provide protection against these whatever they are.
 
-- Executes batch-style swaps that minimize front-running risk and sub-second latency competition, thereby protecting ordinary traders.
+## Batch Execution
 
-  - The order book accumulates incoming limit orders into a batch.
-  - The liquidity module matches accumulated limit orders and orders from the liquidity pool at an equivalent swap price at each batch execution height.
-  - All limit orders in a batch are treated equally and executed at the same swap price.
+The liquidity module uses a batch execution methodology. Deposits, withdrawals, and swap orders are accumulated in a liquidity pool for a pre-defined period that is one or more blocks in length. Orders are then added to the pool and executed at the end of the batch. The size of each batch is configured by using the `UnitBatchSize` governance parameter.
 
-## Features of the Liquidity Module
+## Price Discovery
 
-The main features of the liquidity module are:
+Swap prices in liquidity pools are determined by the current pool coin reserves and the requested swap amount. Arbitrageurs buy or sell coins in liquidity pools to gain instant profit that results in real-time price discovery of liquidity pools.
 
-### Liquidity Pool
+## Escrow Process
 
-A liquidity pool is a coin reserve that contains two kinds of coins in a trading pair. A unique pool exists for each token pair.
+The liquidity module uses a module account that acts as an escrow account. The module account holds and releases the coin amount during batch execution.
 
-A liquidity provider (entity or person) deposits the coins into the liquidity pool and then shares the accumulated swap fees with respect to their pool share. Pool share is represented as possession of pool coins.
+## Refund 
 
-A liquidity pool is permissionless.
+The liquidity module has refund functions when deposit, withdraw, or swap batch states are not successfully executed.
```

Comment:
```suggestion
The liquidity module has refund functions when deposit, withdraw, or swap batch transactions are not successfully executed.
```

Reply:
I think we want "transactions" here... not states?

Reply:
Technically states are right term in this case. Users send transactions and the transactions are accumulated in a batch state.